### ISSUE TITLE
Does not trigger with an initial value

### ISSUE THREAD

**Issue by c-dante** [ExodusMovement/redux-watch#7]
Because of the `var currentValue = getValue(getState(), objectPath)` on initial call to the watch constructor, the change handler that you use to subscribe won't get called until the value has a first change.

I don't want to duplicate my side effect logic by having an initial state, then a watch which performs the same logic, so I end up doing this:

```js
const updateFn = (current, prev) => {
  // ...logic
};
const w = watch(() => someSelector(store.getState()));
const unsub = store.subscribe(w(updateFn));
updateFn(someSelector(store.getState()));
```

It's strange that the current value is captured at the time the watch function is created, but the callback change function isn't run until the next time the state changes.

**Comment by xapple** [comment]
It's been over a year, and no answer. Is this library abandoned ?

**Comment by RyanZim** [comment]
If `updateFn` was called initially, what would be the value of `prev`?

**Comment by c-dante** [comment]
> If `updateFn` was called initially, what would be the value of `prev`?

* Not passed, can use `...args` to capture
* Undefined, ambiguous if you depend on that value
* A third param to indicate the first call
* Can add a second arg to `w(...)` for the initial call, then it's `w(updateFn, initialValue)` (which can be updateFn if your business logic understands)

**Comment by RyanZim** [comment]
> Not passed, can use ...args to capture

Yeah, that isn't great, because you then get `objectPath` as the second parameter

> Undefined, ambiguous if you depend on that value

Right, selectors returning `undefined` is common enough.

> A third param to indicate the first call

I'm not sure I understand this proposal?

> Can add a second arg to w(...) for the initial call, then it's w(updateFn, initialValue)

That's one possibility. Is it crazy to make an initial call only if `initialValue` is provided? Not sure if I like that implicit behavior change.

> which can be updateFn if your business logic understands

Sorry, I'm not understanding this bit

**Comment by marshalokos** [comment]
> It's been over a year, and no answer. Is this library abandoned ?

Oh I’m so sorry for all that happened for the past and now I’m here now and everything will be perfect
